### PR TITLE
OTP Verification feature added

### PR DESCRIPTION
**Pull request by AhmedMinhajEmon**

### CODE REVIEW

File: webapp/webapp/settings.py [hunk 1]
```diff
@@ -89,6 +89,13 @@
         'PORT': 3306,
     }
 }
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_PORT = 587
```
Comment (yiliyavas) [inline webapp/webapp/settings.py:94]: for generating email

File: webapp/main/views.py [hunk 2]
```diff
@@ -57,19 +59,50 @@ def search_posts(request):
...
+    return render(request, 'main/home.html', {'posts': []})  
 def signup(request):
     if request.method == 'POST':
         form = RegisterForm(request.POST)
         if form.is_valid():
-            user = form.save()
-            login(request, user)
-            return redirect('home')
+            user = form.save(commit=False)  # Don't save the user yet
+            otp = random.randint(100000, 999999)  # Generate a 6-digit OTP
```
Comment (yiliyavas) [inline webapp/main/views.py:68]: OTP functions

**Review by yiliyavas — CHANGES_REQUESTED** [review]
aware of those changes

**Comment by yiliyavas** [comment]
done

**Review by AhmedMinhajEmon — COMMENTED** [review]
the function is working fine

**Review by nadinraz — APPROVED** [review]
looks good

**Comment by yiliyavas** [comment]
approve

**Pull request closed without merging**